xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-bridge/src/main/java/org/xwiki/security/authorization/internal/DefaultSecurityEntryReader.java

```diff
@@ -253,17 +253,20 @@ private XWikiDocument getDocument(DocumentReference documentReference) throws Au
 
     /**
      * @param wikiReference the wiki to look for owner
-     * @return a reference to the owner of the wiki
-     * @throws AuthorizationException if the owner could not be retrieved.
+     * @return a reference to the owner of the wiki or {@code null}
      */
     private DocumentReference getWikiOwner(WikiReference wikiReference) throws AuthorizationException
     {
         XWikiContext context = this.xcontextProvider.get();
-        String wikiOwner;
+        String wikiOwner = null;
         try {
             wikiOwner = context.getWiki().getWikiOwner(wikiReference.getName(), context);
         } catch (XWikiException e) {
-            throw new AuthorizationException(wikiReference, "Could not retrieve the owner of this wiki", e);
+            if (e.getModule() != XWikiException.MODULE_XWIKI
+                || e.getCode() != XWikiException.ERROR_XWIKI_DOES_NOT_EXIST) {
```

Comment:
Isn't this enough ?

Reply:
you mean only checking the code and not both module and code? Well probably, but it doesn't hurt to check both I guess?